oca/api.py

```diff
@@ -49,15 +32,66 @@ def method(self):
 
             [1, 2, 3]
 
+        If no container is provided as an argument to the decorator, and a
+        value is returned from the decorated method, a ValueError will be
+        raised.
+
+        .. code-block:: python
+
+            @api.foreach
+            def method(self):
+                # Raises ValueError
+                return 1+2
+
+            @api.foreach
```

Comment:
This would create a fracture in the way that the record-based APIs are used though. The only `api.*` that I can think of that requires a call is `api.returns`, which makes sense to need to call. 

I guess maybe we can just add an assert to make sure that `container` isn't a method in order to avoid me (and others) being stupid.